types/jquery/index.d.ts

```diff
@@ -25,14 +25,14 @@
 // TypeScript Version: 2.3
 
 declare module 'jquery' {
-    function factory(window: Window): JQueryStatic;
+    function factory(window: Window): JQueryStatic & JQuery;
```

Comment:
I'm considering changing the signature of `factory` to:
```typescript
function factory(window: Window, discriminator: boolean): JQueryStatic;
```
This is technically wrong but the extraneous argument should be ignored by the factory method.

From https://github.com/jquery/jquery/blob/3.2.0/src/wrapper.js#L30:
```javascript
function( w ) {
    if ( !w.document ) {
        throw new Error( "jQuery requires a window with a document" );
    }
    return factory( w );
};
```

Consumers will have to know to pass in the extra parameter in order to get correct typing. I get the impression that users who use the factory are a small proportion of users. This is a compromise solution that tries to give a good and intuitive experience to the majority of users. I haven't been able to come up with a solution that doesn't compromise in one way or another.